docs/ocis/development/debugging.md

```diff
@@ -179,6 +179,171 @@ For example `deployments/examples/ocis_wopi/docker-compose.yml`
 },
 ```
 
+### Debugging the Collaboration stack
+Sometimes for debugging purposes, we want to run the WOPI stack with the collaboration service locally.
+Let's focus on an example that uses the local runs of the ocis and collaboration together with the containerized traefik and collabora.
+Please note: Against the stack that uses https://github.com/cs3org/wopiserver, we don't need the app_provider anymore. The new collaboration service now includes an app_provider.
+
+
+1. Update the hosts file
+```bash
+127.0.0.1 ocis.owncloud.test
+127.0.0.1 traefik.owncloud.test
+127.0.0.1 collabora.owncloud.test
+127.0.0.1 collaboration.owncloud.test
+```
+
+2. Update the ocis/services/proxy/pkg/config/csp.yaml
+``` yaml
+  frame-src:
+    - '''self'''
+    - 'https://embed.diagrams.net/'
+    - 'https://collabora.owncloud.test/'
+  img-src:
+    - '''self'''
+    - 'data:'
+    - 'blob:'
+    - 'https://collabora.owncloud.test/'
+```
+
+
+3. Run docker compose
+
+OCIS_DOMAIN=ocis.owncloud.test:9200
+
+
+4. Run ocis
+``` bash
+OCIS_URL=https://ocis.owncloud.test:9200 \
+PROXY_ENABLE_BASIC_AUTH=true \
+MICRO_REGISTRY=nats-js-kv \
+MICRO_REGISTRY_ADDRESS="127.0.0.1:9233" \
+NATS_NATS_HOST="0.0.0.0" \
+NATS_NATS_PORT=9233 \
+GATEWAY_GRPC_ADDR="0.0.0.0:9142" \
+APP_PROVIDER_EXTERNAL_ADDR="com.owncloud.api.app-provider" \
+APP_PROVIDER_DRIVER="wopi" \
+APP_PROVIDER_WOPI_APP_NAME="FakeOffice" \
+APP_PROVIDER_WOPI_APP_URL=http://localhost:8080 \
+APP_PROVIDER_WOPI_INSECURE=true \
+APP_PROVIDER_WOPI_WOPI_SERVER_EXTERNAL_URL=http://localhost:9300 \
+APP_PROVIDER_WOPI_FOLDER_URL_BASE_URL=https://localhost:9200 \
```

Comment:
Will these env configs have any effects? because with or without these configs I don't see any difference in the webUI.
with or without these configs, I can use collabora office.

Reply:
I also tried `APP_PROVIDER_WOPI_APP_ICON_URI: https://collabora.owncloud.test/favicon.ico` to get the icon but icon didn't work
![Screenshot from 2024-05-22 15-14-13](https://github.com/owncloud/ocis/assets/52366632/801137d2-d660-439b-ae26-c1848d534b89)

Reply:
You are right. This envs are not needed.
Added COLLABORATION_APP_ICON=https://collabora.owncloud.test/favicon.ico